shared/cert.go

```diff
@@ -490,7 +490,7 @@ func GetRemoteCertificate(address string, useragent string) (*x509.Certificate,
 	}
 
 	// Connect
-	req, err := http.NewRequest("GET", address, nil)
+	req, err := http.NewRequest("HEAD", address, nil)
```

Comment:
@simondeziel I'd prefer functional changes like this aren't included in PR that it isnt related to ("Improved validation of project and instance names").

If the remote doesnt support HEAD requests then this could be a regression.

Some things that should be considered before we land this change (if not already done):

- How far back in terms of LXD server versions did you test this?
- Is this function only ever called for LXD remotes or can it be used for anything else?
- Is it used in microcluster/micro* and do their endpoints also support HEAD?

Reply:
I've dropped it, not related indeed.

Those are going point regarding compatibility and I admit making the assumption that HEAD would be permitted/support anywhere GET is. Probably not a given.

Since we are only interested at the TLS layer, I concluded the upper HTTP protocol didn't matter much. That was wrong, as in case of error, we stop before peaking at the TLS info we obtained.

Reply:
> @simondeziel I'd prefer functional changes like this aren't included in PR that it isnt related to ("Improved validation of project and instance names").

I note you specifically said "functional changes" which I think is very reasonable to keep separated and I'll try to improve that in the future, thanks for pointing that out!

I am often tempted to include minor changes (fmt.Sprintf replacement) like I did in this PR too. Those unfortunately sometimes cause issues as the team has noticed and cleaned up after myself :/

Should I also keep those "minor/drive-by" fixes for separated PRs?  Would that make it easier to review/catch mistakes?

Reply:
yes probably, it also makes it easier to get the other PRs merged to keep moving things forward.